controllers/articles.js

```diff
@@ -0,0 +1,247 @@
+
+      const queryArticle = await article.create({
+        title, body, author, slug, description, status: flag, tagList: tags,
+      });
+      return res.status(201).send({
+        status: res.statusCode,
+        article: {
+          id: queryArticle.id,
+          title: queryArticle.title,
+          description: queryArticle.description,
+          body: queryArticle.body,
+          slug: queryArticle.slug,
+          tags: [queryArticle.tagList]
+        }
+      });
+    } catch (err) {
+      if (err.message) {
+        res.status(500).send({
+          error: 'Your are not an author yet, please activate your account.',
+        });
+      }
+    }
+  },
+
+  /*
+   * updating a post based to its slug.
+   *
+   * For updating an article, status flag has to be passed
+   * into the request body object otherwise the article will be in drafted.
+   */
+  updateAnArticle: async (req, res) => {
+    try {
+      const oldSlug = req.params.slug;
+      const { title, body, tagList } = req.body;
+      const slug = `${title.toLowerCase().split(' ').join('-').substring(0, 20)}${crypto.randomBytes(5).toString('hex')}`;
+      const description = body.substring(0, 100);
+
+      await article.update(
+        {
+          title, body, slug, description, tagList
+        },
+        {
+          where: { slug: oldSlug }
+        }
+      );
+      const updateThisArticle = await article.findOne({ where: { slug } });
+      if (!updateThisArticle) {
+        return res.status(404).send({
+          status: res.statusCode,
+          errorMessage: 'Article not found, please create a new article instead',
+        });
+      }
+      return res.status(200).send({
+        status: res.statusCode,
+        article: {
+          title: updateThisArticle.title,
+          body: updateThisArticle.body,
+          slug: updateThisArticle.slug,
+          tagList: updateThisArticle.tagList,
+        }
+      });
+    } catch (err) {
+      if (err.message) {
+        res.status(500).send({
+          status: res.statusCode,
+          errorMessage: 'No article to update, please create an article first',
+        });
+      }
+    }
+  },
+
+  /*
+   * Deleting a post based to its slug.
+   *
+   * For avoiding permanent deletion of the content from the database
+   * we set the flag to the article that it is deletes.
+   */
+  deleteArticle: async (req, res) => {
+    const { slug } = req.params;
+    const row = await article.update({ deleted: 1 }, { where: { slug, deleted: 0 } });
+    if (row[0] === 0) {
+      return res.status(404).send({
+        status: res.statusCode,
+        message: 'No article found for this slug',
+      });
+    }
+    return res.status(200).send({
+      status: res.statusCode,
+      message: 'Article deleted successfully',
+    });
+  },
+
+  /*
+  * Retieving all the published articles based to the author
+  * and the status of the article (Published).
+  */
+  getAllPublished: async (req, res) => {
```

Comment:
rename this and similar names to `getAllPublishedArticles`, `getAllDraftArticles` etc.
Variable names should be self describing so that people reading or debugging your code can make sense without having to ask you what everything means.